waku/v2/protocol/store/waku_store.go

```diff
@@ -374,31 +374,35 @@ func (store *WakuStore) storeIncomingMessages(ctx context.Context) {
 
 func (store *WakuStore) onRequest(s network.Stream) {
 	defer s.Close()
-
+	logger := store.log.With(logging.HostID("peer", s.Conn().RemotePeer()))
 	historyRPCRequest := &pb.HistoryRPC{}
 
 	writer := protoio.NewDelimitedWriter(s)
 	reader := protoio.NewDelimitedReader(s, math.MaxInt32)
 
 	err := reader.ReadMsg(historyRPCRequest)
 	if err != nil {
-		store.log.Error("error reading request", err)
+		logger.Error("reading request", zap.Error(err))
 		metrics.RecordStoreError(store.ctx, "decodeRPCFailure")
 		return
 	}
-
-	store.log.Info(fmt.Sprintf("%s: Received query from %s", s.Conn().LocalPeer(), s.Conn().RemotePeer()))
+	logger = logger.With(zap.String("id", historyRPCRequest.RequestId))
+	if query := historyRPCRequest.Query; query != nil {
+		logger = logger.With(logging.Filters(query.GetContentFilters()))
+	}
+	logger.Info("received query")
 
 	historyResponseRPC := &pb.HistoryRPC{}
 	historyResponseRPC.RequestId = historyRPCRequest.RequestId
 	historyResponseRPC.Response = store.FindMessages(historyRPCRequest.Query)
 
+	logger = logger.With(zap.Int("messages", len(historyResponseRPC.Response.Messages)))
 	err = writer.WriteMsg(historyResponseRPC)
 	if err != nil {
-		store.log.Error("error writing response", err)
+		logger.Error("writing response", zap.Error(err), logging.PagingInfo(historyResponseRPC.Response.PagingInfo))
```

Comment:
The paging info is a somewhat complex value, but was intended to help with debugging query results. In our fork we only include it in debug mode (I didn't include that facility in my proposal, since that was a bit more obscure bit), here it would be included with all response write errors, which may not be a great fit. Again happy to drop it, just let me know.